docs: lazy loading link update

nikvarma opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

MeAkib committed:
docs(docs-infra): cleanup repeated scss from typography (#51579)

PR Close #51579

bampakoa committed:
docs: update Aristeidis Bampakos bio (#51779)

PR Close #51779

wojtek1150 committed:
docs: Update update-to-version-16.md (#51857)

Fix example for required inputs that leads to TS2564 error - property has no initializer

PR Close #51857

sdedieu committed:
docs(docs-infra): add if cases in templates to avoid internal marked parameters or methods to appear in the documentation (#52020)

Fix case of not wanted internal properties displayed in the documentation by adding if blocks in the concerned templates. So if the option is marked as internal the template won't create an entry in the documentation table for the concerned property of method

Fixes #52018

PR Close #52020

mhagnumdw committed:
InMemoryBackendConfigArgs.passThruUnknownUrl: fix tsdoc (#52273)

PR Close #52273

Commit:
docs: add new component imports documentation structure (#52282)

PR Close #52282

AnkitaSood committed:
docs: add ankita (#52319)

add Ankita as GDE in contributors.json

PR Close #52319

JeanMeche committed:
docs: add 2 new diagnostics created recently. (#52329)

PR Close #52329

iliasaddab committed:
docs: update i18n-common-merge.md (#52400)

correct spelling

PR Close #52400

JeanMeche committed:
refactor(platform-browser): remove non necessary factories (#52439)

Let's keep the code a simple as necessary

PR Close #52439

JeanMeche committed:
refactor(platform-browser): remove the depency on `window` for the `AngularProfiler`. (#52440)

`performance` and `console` are both available on NodeJS, we can replace `window` with `globalThis`.

PR Close #52440

pablopaul committed:
docs: update start-data.md (#52577)

PR Close #52577

AhsanAyaz committed:
docs(changelog): release notes, fix the typos etc (#52628)

fixed the typos in a few places
added a missing backtick
removed an unnecessary bullet point

PR Close #52628

mattezell committed:
docs: Corrected link for defer related blocks (#52704)
PR Close #52704

nelsongutidev committed:
docs: add correct syntax highlighting type in event binding documentation (#52712)

PR Close #52712

nelsongutidev committed:
docs: add correct language for syntax highlighting on new docs (#52712)

PR Close #52712

amilamen committed:
docs: bad redirection in core Injector page for providers and injection token links (#52775)
PR Close #52775

sasidharansd committed:
docs: fix invalid file replacement configuration (#52802)

PR Close #52802

sdedieu committed:
docs: change the file of the provideHttpClient standalone example to match cli generated project (#52806)

The "Providing HttpClient through dependency injection" documentation now shows that in case of a standalone boostraped app, the provideHttpClient is to include in the app.config.ts file.

Also updated the http setup server communication example to align the new doc (adev) with the old one (aio).

PR Close #52806

ranma42 committed:
refactor: fix a number of typos throughout the codebase (#52826)

Fix several typos, mostly in the `adev` site.

PR Close #52826

SahulKola committed:
fix: cta clickability issue in adev homepage. (#52905)
PR Close #52905

lsst25 committed:
docs(docs-infra): close tag in the example of Composing Components tutorial (#52923)

PR Close #52923

arturovt committed:
fix(animations): cleanup DOM elements when root view is removed with async animations (#53033)

Currently, when using `provideAnimationsAsync`, Angular uses `AnimationRenderer`
as the renderer. When the root view is removed, the `AnimationRenderer` defers the actual
work to the `TransitionAnimationEngine` to do this, and the `TransitionAnimationEngine`
doesn't actually remove the DOM node, but just calls `markElementAsRemoved()`.

The actual DOM node is not removed until `TransitionAnimationEngine` "flushes".

Unfortunately, though, that "flush" will never happen, since the root view is being
destroyed and there will be no more flushes.

This commit adds `flush()` call when the root view is being destroyed.

PR Close #53033

oliv37 committed:
docs: fix add builder from package (#53154)

PR Close #53154

ilijaleko committed:
docs: fix-build-path-in-http-server-command (#53226)

PR Close #53226

JeanMeche committed:
refactor(devtools): Add support for signals. (#53269)

The devtools now support signals.
Writable signals of primitives are editable.
Object Signal and other non-writable signals (like computed) are not editable.

Co-authored-by: Tomasz Ducin <tomasz.ducin@gmail.com>

PR Close #53269

srinathsree66 committed:
docs: fixed req.method in http client testing (#53395)
PR Close #53395

srinathsree66 committed:
docs: fix router api url link (#53398)
PR Close #53398

AdityaK60 committed:
docs: Fix router api url link on tutorial page (#53416)

Signed-off-by: Aditya Kumar <aditya.kumar60@infosys.com>

PR Close #53416

AdityaK60 committed:
docs: Fix minor grammatical fixes in the documentations (#53418)

Signed-off-by: Aditya Kumar <aditya.kumar60@infosys.com>

PR Close #53418

Commit:
refactor(devtools): improving type safety (#53436)

This PR reduces the number of unnecessary `any` occurrences in devtools packages.

PR Close #53436

rossyman committed:
docs: fix nonsensical injection provider docs (#53456)

PR Close #53456

nickolashkraus committed:
docs: With >>> with (#53472)

PR Close #53472

sasidharansd committed:
docs: fix broken code example (#53531)

PR Close #53531

pmvald committed:
refactor(compiler-cli): add API for extra import tracker (#53543)

The tracker is responsible for registering the extra imports during the analysis and resolve compiler phases, and later to be used by the transformer to get a list of extra imports to be generated for each source file.

This commit only contains the API, and the actual implementation for each method will be done in subsequent commits where an application of that method is available and so tests can be written for the implementation.

PR Close #53543

pmvald committed:
refactor(compiler-cli): incorporate `LocalCompilationExtraImportsTracker` in the compilation workflow (#53543)

This commit includes a skeleton of how the tool `LocalCompilationExtraImportsTracker` is used in the overall compilation workflow end-to-end.

First of all, a new option `generateExtraImportsInLocalMode` is added, whose presence will make `LocalCompilationExtraImportsTracker` part of the compilation process. When this option is set an instance of `LocalCompilationExtraImportsTracker` is created within the NgCompiler. Then it is passed to the Ivy transformer and plumbed all the way down and the extra imports registered in it are added to the `ImportManager` instances before the imports are added from `ImportManager` to the generated file. This required adding a new method `generateSideEffectImport` to the `ImportManager`, which is an empty method and will be implemented in the subsequent commits.

This commit expected to make no change in the compilation behavior as the methods are not implemented yet.

PR Close #53543

pmvald committed:
refactor(compiler-cli): ability to generate side effect imports (#53543)

As the first step,  the import manager's `generateSideEffectImport` method is implemented to enable it to store info for side effect imports. Next, the helper `addImports` is modified to be able to generate correct statement for side effect imports.

These changes will be tested in the subsequent commits when these tools are used to generate an actual extra import for the generated file.

PR Close #53543

pmvald committed:
feat(compiler-cli): generate global imports in local compilation mode (#53543)

With option `generateExtraImportsInLocalMode` set in local compilation mode, the compiler generates extra side effect imports using this rule: any external module from which an identifier is imported into an NgModule will be added as side effect import to every file in the compilation unit. To illustrate this better assume the compilation unit has source files `a.ts` and `b.ts`, and:

```
// a.ts
import {SomeExternalStuff} from 'path/to/some_where';
import {SomeExternalStuff2} from 'path/to/some_where2';

...

@NgModule({imports: [SomeExternalStuff]})

```

then the extra import `import "path/to/some_where"` will be added to both `a.js` and `b.js`. Note that this is not the case for `import "path/to/some_where2"` though, since the symbol `SomeExternalStuff2` is not imported into any NgModule.

The math behind this mechanism is, in local compilation mode we cannot resolve component external dependencies fully. For example if a component in `a.ts` uses an external component defined in an external file `some_external_comp.ts` then we can generate the import to this file in `a.js`. Instead, we want to generate an import to a file that "gurantees" that `a.js` is placed after `some_external_comp.js` in the bundle. Now since the component in  `some_external_comp.ts` is used in `a.ts`, then there must be a chain of imports starting from the NgModule that declares the component in `a.ts` to the component in `some_external_comp.ts`. This chain means some file in the same compilation unit as `a.ts` should import some external NgModule which includes `some_external_comp.ts` in its transitive closure and import it to some NgModule. So by adding this import to `a.js` we ensure that the bundling will have the right order.

PR Close #53543

pmvald committed:
refactor(bazel): make option generateExtraImportsInLocalMode configurable (#53543)

This is to allow testing this option using bazel

PR Close #53543

pmvald committed:
refactor(compiler-cli): run resolve phase fully for components in local mode when `generateExtraImportsInLocalMode` is set (#53543)

In this commit the resolve method for components is run fully when the option `generateExtraImportsInLocalMode` is set. This is because we need local component depedencies in order to generate extra imports causing by them. This requires cutting some resolve phase logics that are unnecessary in local mode, such as diagnostics.

PR Close #53543

pBouillon committed:
docs: fix reference to forwardRef in ForwardRefFn JSdoc (#53556)

PR Close #53556

pmvald committed:
feat(compiler-cli): generate extra imports for component local dependencies in local mode (#53543)

With option `generateExtraImportsInLocalMode` set, in local mode the compiler generates extra imports for each component local dependencies. Here local dependencies means all component's dependencies within the same compilation unit.

To achieve this, the compiler performs a "local version" of its regular static analysis to find each component's deps, and these deps are used to generate extra side effect imports.

PR Close #53543

pmvald committed:
fix(compiler-cli): generating extra imports in local compilation mode when cycle is introduced (#53543)

At the moment the extra import generation in local compilation mode fails if these extra imports produce a cycle. To handle this, the cycle handling strategy is updated for local compilation, and following the behaviour in the full compilation mode, the compiler does not generate extra import if it leads to cycle and instead leave things to the runtime.

PR Close #53543

ahhmoh committed:
docs: testing-components-basics.md and components-basics.md links to css selector documentation are dead (#53586)

docs: testing-components-basics.md and components-basics.md links to css selector documentation are dead

As MDN's documentation link to the "css selector" topic is dead, this commit makes the angular documentation use the new valid link.

PR Close #53586

trungvose committed:
docs: add Trung Vo to GDE resources (#53599)

PR Close #53599

atscott committed:
refactor(router): Update integration tests to cover navigation and history API (#53799)

This commit updates the router integration tests to cover both the
classic History and the new Navigation API. There is more work to be
done here, but this commit works to prove the efficacy of the
`FakeNavigation` implementation.

PR Close #53799

AdityaK60 committed:
docs: Fix open link in new tab (#53647)

Signed-off-by: Aditya Kumar <aditya.kumar60@infosys.com>

PR Close #53647

angular-robot committed:
docs: update events (#53654)

Generated `events.json` with the latest events retrieved from the Firebase DB.

Closes #53229

PR Close #53654

louaymeskini committed:
docs: Fix links for NgZone (#53667)

PR Close #53667

jivanf committed:
docs: Use correct service name in hierarchical dependency injection guide (#53723)

This commit replaces `TitleService` with `TiresService` in the specialized providers scenario described in the guide.
PR Close #53723

Ahmed-Hakeem committed:
docs: focus on the respective card of the current active line while scrolling (#53733)

make the active cards have the focus while scrolling like their
respective lines in the scroll service

PR Close #53733

JeanMeche committed:
refactor(compiler): remove mentions of unused compiler options. (#53746)

Those options where removed in #49672.

PR Close #53746

JeanMeche committed:
refactor(core): remove `InjectionToken` descriptions in optimized builds. (#53747)

We started guarding the `InjectionToken` descriptions with `ngDevMode`. Let's generalize that accross the FW.

PR Close #53747

Magda98 committed:
docs(docs-infra): move comma in Angular Language Service section (#53759)

PR Close #53759

jase88 committed:
docs: fix http handler usage on interceptor example (#53772)
PR Close #53772

atcastle committed:
feat(common): add placeholder to NgOptimizedImage (#53783)

Add a automatic placeholder implementation supporting loader-based and data URL placeholders

PR Close #53783

pkozlowski-opensource committed:
refactor(core): introduce query-as-signal authoring functions (#53829)

This commit adds signatures of the quer-as-signal authoring functions
and their respective type tests.

PR Close #53829

pkozlowski-opensource committed:
test(core): support type tests for different authoring functions (#53829)

The new type testing infrastructure was introduced for the input-as-signals
authoring functions. This commit modifies this infrastructure to make it more
generic and support queries-as-signals.

PR Close #53829

macvag committed:
docs: fix broken hyperlink in tutorials first app search (#53816)

PR Close #53816

devversion committed:
refactor(core): separate `InputSignal` and `InputSignalWithTransform` (#54053)

This commit separates `InputSignal` for input signals with transforms.
The reason being that most of the time, signal inputs are not using
transforms and the generics are rather confusing.

Especially for users with inferred types displayed in their IDEs, the
input signal types are seemingly complex, even if no transform is used.

For this reason, we are introducing a new type called
`InputSignalWithTransform`. This type will be used for inputs with
transforms, while non-transform inputs just use `InputSignal`.

A notable fact is that `InputSignal` extends `InputSignalWithTransform`,
with the "identity transform". i.e. there is no transform. This allows
us to share the code for input signals. In practice, we don't expect
users to pass around `InputSignal`'s anyway.

PR Close #54053

gleb-28 committed:
docs: removed repeated words "when" (#53828)

Fixes #53827

PR Close #53828

sasidharansd committed:
docs: fix spelling errors in tutorials steps (#53840)

PR Close #53840

atscott committed:
refactor(core): node removal notifies scheduler only when animations are enabled (#53857)

Node removal is immediate and does not require change detection to run
when animations are not provided. This refactor makes the animation
engine notify the scheduler rather than doing it on all node removals.

PR Close #53857

josephperrott committed:
ci: set up previews of adev (#53941)

Set up preview actions to allow for a preview of the adev site to be deployed for a PR with the `adev: preview` label.

PR Close #53941

hybrist committed:
refactor(core): Allow mutation instead of reassignment of ngDevMode (#53862)

In some bundling scenarios, there may be local references to `ngDevMode` that need to be kept in sync with the global variable. This becomes hard to impossible if the global is reassigned. This allows setting the global to an empty object instead of `true` and preserve identity during `initNgDevMode`.
PR Close #53862

josephperrott committed:
build(docs-infra): enable prerendering for adev (#53871)

Enable the prerender option for adev's angular.json file.

PR Close #53871

pmvald committed:
feat(compiler-cli): support host directives for local compilation mode (#53877)

At the moment local compilation breaks for host directives because the current logic relies on global static analysis. This change creates a local version by cutting the diagnostics and copying the directive identifier as it is to the generated code without attempting to statically resolve it.

PR Close #53877

angular-robot committed:
docs: lazy loading link update

cexbrayat committed:
refactor(compiler-cli): interpolatedSignalNotInvoked diagnostic for input signals (#53883)

This updates the extended diagnotic to handle input signals as well.

PR Close #53883

nikvarma committed:
docs: typo Chat to chart fix (#53892)

PR Close #53892

devversion committed:
fix(compiler-cli): do not throw fatal error if extended type check fails (#53896)

Currently when the extended type check fails due to an import reference
that cannot be generated, the fatal diagnostic is not caught and
not properly exposed as a `ts.Diagnostic` that can be gracefully
handled. This is inconsistent to non-extended type checking diagnostics.

This is problematic because Angular CLI applications currently fail in
obscure ways because:

- the CLI does not expect `getDiagnosticsForFile` to actually throw
  runtime errors.
- the CLI does not seem to properly print these errors given the
  parallel workers and build excection, and those errors are
  especially hard to debug because there is no `stack` for
  `FatalDiagnosticError`'s.

Example: `MyDir` is not exported and the type check block cannot reference it.

PR Close #53896

AndrewKushnir committed:
refactor(compiler): extra diagnostics for `@defer` in local compilation mode (#53899)

This commit adds extra logic to produce a diagnostic in case `@Component.deferredImports` contain types from imports that also bring eager symbols. This would result in retaining a regular import and generating a dynamic import, which would not allow to defer-load dependencies.

PR Close #53899

AndrewKushnir committed:
refactor(compiler): enable `register` and `resolve` phases for local compilation (#53901)

This commit update the logic to enable `register` and `resolve` phases for local compilation. Those phases will be useful for local compilation in certain cases (will be used in followup PRs).

PR Close #53901

JeanMeche committed:
docs: add `platform-browser/animations/async` entry to AIO (#53914)

fixes #53912

PR Close #53914

ilirbeqirii committed:
build(devtools): prevent underscores from being added at build time (#53921)

Add the keepNames config property to esbuild in order to prevent prepending of underscores to the class names.

PR Close #53921

pkozlowski-opensource committed:
refactor(core): re-organize queries code (#53922)

This commit splits the query implementation and instructions
into a separate files. This is a pattern frequently used by
other functional areas of the framework and is a preparation for
introducing queries-as-signals where we are going to see more
instructions delegating to the same core functionality.

PR Close #53922

JeanMeche committed:
refactor(common): Use Nullish coalescing assignment when possible. (#53931)

Inspired by #53923, lets simplify code when possible.

PR Close #53931

JeanMeche committed:
refactor(router): Use Nullish coalescing assignment when possible. (#53930)

Inspired by #53923, lets simplify code when possible.

PR Close #53930

pmvald committed:
feat(compiler-cli): make it configurable to generate alias reexports (#53937)

At the moment when unified host is selected (through option `_useHostForImportGeneration`) the compiler always generates alias reexports. Such reexports are mainly generated to satisfy strict dependency condition for generated files. Such condition is no longer the case for G3. At the same time, these alias reexports make it impossible to mix locally compiled targets with globally compiled targets. More precisely, a globally compiled target may not be able to consume a locally compiled target as its dependency since the former may import from the alias reexports which do not exist in the latter due to local compilation mode. So, to make global-local compilation interop possible, it is required to be able to turn off alias reexport generation.

PR Close #53937

kabrunko-dev committed:
refactor(common): simplify date pipe metadata (#53939)

Remove pure property from date pipe decorator

PR Close #53939

devversion committed:
refactor(compiler-cli): support generation of signal-based queries (#53978)

This commit introduces the compiler output generation for signal-based
queries. Signal-based queries will have new creation-mode instructions
and update instructions to advance the current query indices in the
global shared context.

An output like the following is the expected output for signal-based
queries:

```
i0.ɵɵdefineComponent({
  viewQuery: function App_Query(rf, ctx) {
      if (rf & 1) {
          i0.ɵɵviewQuery(ctx.d, _c0, 5);
          i0.ɵɵviewQuerySignal(ctx.ds1, _c0, 5);
          i0.ɵɵviewQuerySignal(ctx.ds2, _c0, 5);
      }
      if (rf & 2) {
          let _t;
          // only change-detected queries need explicit refresh
          i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.d = _t.first);
          // we bump up current query index by 2 positions since there are 2 signal-based queries
          i0.ɵɵqueryAdvance(2);
      }
      …
  },
  …
});

```

Note: For now, the collapsing of multiple advance instructions is not
implemented. This will be a follow-up.

Note 2: A couple of query helpers are now in their own file. This makes
it easier to focus on query-specific compiler code. The new function is
called `createQueryCreateCall`, which is a modified variant of the
existing function that previously only generated query parameters.

PR Close #53978

kabrunko-dev committed:
refactor(common): simplify i18n plural and select pipe metadata (#53939)

Remove pure property from pipe decorator

PR Close #53939

devversion committed:
refactor(compiler-cli): expose helper for recognizing initializer APIs (#53978)

The new `input` API is recognized using class member initializers.
We want to support similar APIs for queries, using e.g. `viewChild`
or `viewChild.required`.

This commit extracts the input recognition API and makes it reusable,
so that the same logic can be used to detect queries on a class member.

Additional changes:
  - replacing `coreModule` with the simpler `isCore` parameter. This is
    more readable.
  - support for detecting a list of API names on a single class member.
    This allows us to detect possible query functions on the same class
    member without having to check X times. We simply check for the
    initializer API pattern and check if one API function name matches.

PR Close #53978

ilirbeqirii committed:
docs: fix type at change detection with pipes section (#53940)

Fix a typo of word 'heroes' at docs guide for change detection with pure pipes

PR Close #53940

josephperrott committed:
ci: run adev tests on CI (#53941)

Run tests in adev for CI

PR Close #53941

puckowski committed:
feat(compiler): scope selectors in @starting-style (#53943)

make sure selectors inside @starting-style queries are correctly scoped

PR Close #53943

josephperrott committed:
refactor(devtools): migrate devtools to prettier formatting (#53945)

Migrate formatting to prettier for devtools from clang-format

PR Close #53945

josephperrott committed:
refactor: migrate tools to prettier formatting (#53947)

Migrate formatting to prettier for tools directory from clang-format

PR Close #53947

AleksanderBodurri committed:
docs(devtools): remove devtools/CONTRIBUTING.md (#53948)

This is an artifact left over form the repo merge. There is already an existing CONTRIBUTING.md for this repo

PR Close #53948

AleksanderBodurri committed:
docs(devtools): update development docs (#53948)

These docs are out of date with how we currently build devtools in dev mode and as a browser extension. This commit brings these docs up to date.

PR Close #53948

josephperrott committed:
refactor: migrate modules to prettier formatting (#53954)

Migrate formatting to prettier for modules directory from clang-format

PR Close #53954

josephperrott committed:
ci: use the config from adev for adev previews (#53955)

Properly use the adev firebase config.

PR Close #53955

devversion committed:
refactor(compiler-cli): recognize signal-based queries in class initializers (#53978)

This commit uses the initializer API recognition that we built for
signal-based inputs, and teaches the compiler to recognize class members
that refer to `viewChild`, `viewChildren`, `contentChild` or
`contentChildren`. Those will declare signal-based view or content queries.

PR Close #53978

thePunderWoman committed:
fix(core): Change defer block fixture default behavior to playthrough (#53956)

This inverts the default behavior of test bed to use playthrough for defer blocks instead of manual.

fixes: #53686

PR Close #53956

josephperrott committed:
ci: use ng-dev-previews project for previews (#53958)

Rather than use the actual adev site, we should be using the ng-dev-preview site

PR Close #53958

alan-agius4 committed:
docs: update docs to reflect changes in Angular CLI version 17.1 (#53949)

This changes include mentioning the experimental application builder migration and changes to the `outputPath` and `optimization` option.

PR Close #53949

josephperrott committed:
ci: provide proper location for artifacts to allow for firebase to find the artifacts for deployment (#53960)

Move the artifacts during deployment into adev/ so that its at the same level as the firebase.json file.

PR Close #53960

dylhunn committed:
Revert "refactor(core): improve forwardRef typings (#53880)" (#53961)

This reverts commit af6f6e6448edffcd644b0bb6ce155c3ca4270971.

PR Close #53961

dylhunn committed:
revert: no longer depend on `forwardRef` changes (#53961)

Revert only the part of the signal queries PRs that depends on the forward ref changes, which were reverted.

PR Close #53961

dylhunn committed:
docs: release notes for the v17.1.0 release

atscott committed:
fix(compiler): Update type check block to fix control flow source mappings (#53980)

The source mappings and types for various pieces of the control flow
were not quite right and prevented the language service from providing
accurate information.

fixes https://github.com/angular/vscode-ng-language-service/issues/1988

PR Close #53980

pmvald committed:
refactor(compiler-cli): run register phase in local compilation mode (#53543)

When option `generateExtraImportsInLocalMode` is set, we need to compute component local depednecies in order to generate extra imports related to them. At the same time running the register phase in general is harmless in local compilation. So we run it anyway.

PR Close #53543

AndrewKushnir committed:
refactor(compiler-cli): rephrase an error message related to `@defer` and local compilation (#54030)

This commit updates the error message to cover cases when imported symbols have eager references inside of a file.

PR Close #54030

angular-robot committed:
build: update cross-repo angular dependencies (#53970)

See associated pull request for more information.

PR Close #53970

devversion committed:
refactor(core): correct types for queries where locator was not split at compile time (#53978)

Currently, Angular tries to recognize string locators/predicates for
queries at compile time, and attempts to split multi-selector predicates
into an array as generated output. This is a a performance optimization.

In practice, this works most of the time because the compiler can detect
string locactors/predicates through static analysis.

Though, there are cases where it's not possible. That is when advanced
constructs are used, identifier references etc. that ultimately evaluate
to a string. Currently this breaks with queries and also surfaces now
with signal-based queries.

PR Close #53978

devversion committed:
test: add compiler integration test for signal-based queries (#53978)

Adds a compiler integration test for recognizing signal-based queries,
and emitting the expected output. Concrete output will be verified via
the compliance tests.

PR Close #53978

devversion committed:
refactor(compiler-cli): generate partial compilation output for signal-based queries (#53978)

This commit ensures that libraries can use signal-based queries, and the
partial compilation output will capture their metadata.

The linker is updated to support parsing this.

Two notes:

1. Older linker versions are not capable of parsing this, so the minimum
   version for signal-based queries is adjusted when such are used.
2. We only emit `isSignal` metadata for queries when signal queries are
   used. This enables libraries to continue supporting older linker
   versions, if signal-based queries are not used.

PR Close #53978

devversion committed:
test: add compliance tests for signal-based queries (#53978)

This commit adds compliance tests to ensure that the generated output of
signal-based queries matches our expectation.

Note: collapsing query advance instructions is not implemented yet.

PR Close #53978

angular-robot committed:
build: update all non-major dependencies (#53979)

See associated pull request for more information.

PR Close #53979

thePunderWoman committed:
docs: updates defer guide to reflect updated default defer block test behavior (#53975)

This updates the deferrable view guide to specify playthrough as the default with examples of configuring to manual.

PR Close #53975

josephperrott committed:
refactor: migrate scripts to prettier formatting (#53976)

Migrate formatting to prettier for scripts from clang-format

PR Close #53976

josephperrott committed:
refactor: migrate animations to prettier formatting (#53977)

Migrate formatting to prettier for animations from clang-format

PR Close #53977

atscott committed:
refactor(core): Update `ComponentFixture` behavior when using zoneless scheduler (#54024)

When the zoneless scheduler is provided, we want to update the behavior
of `ComponentFixture` to address common issues and painpoints in testing.
Developers should never have to call `detectChanges` on a fixture
manually. Instead of calling `detectChanges` after performing an
action that updates state and requies a template refresh, developers
should wait for change detection to run because the update needs to also have
notified the scheduler. If this was not the case, the component would
not work correctly in the application. Calling `detectChanges` to force
an update could hide real bugs.

This commit also updates the zoneless tests to uses `ComponentFixture`
instead of manually attaching to the `ApplicationRef` and rewriting a
lot of the helpers (`getDebugNode`, `isStable` as a value, `whenStable` as a
Promise).

PR Close #54024

pmvald committed:
fix(compiler-cli): show proper error for custom decorators in local compilation mode (#53983)

At the moment local compilation mode does not support custom decorators, and it leads to unhandled errors. In this change a compile time diagnostic is produced in local mode for custom decorators. This is a temporary solution since there are few custom decorators are in use in g3. Custom decorators will be eventually supported in local mode.

PR Close #53983

devversion committed:
build: update g3 sync config to match internal one (#53987)

There were some changes to the Copybara config that should
be reflected here as well.

PR Close #53987

devversion committed:
refactor(compiler-cli): move more query generation logic to dedicated file (#54019)

Without any modifications, this commit moves more of the the query
generation logic into its dedicated file.

PR Close #54019

devversion committed:
refactor(compiler-cli): collapse multiple query advance statements (#54019)

Collapses multiple sibling query advance statements into single
query advance invocations. This will help reducing generated code
for directives/components with many queries.

PR Close #54019

devversion committed:
refactor(core): improve safety of input/query initializer API detection (#54019)

Due to some refactorings, we were only checking the function name
and whether it originates from an import. We should also verify the
module. This seems like logic we lost in the refactorings.

PR Close #54019

JeanMeche committed:
build: Adding myself for devtools approval (#53989)

This to help Alex with the reviewing of devtools PRs.

PR Close #53989

therealdennis08 committed:
docs: correct grammar error on start page (#53993)

Fixed grammatical error

PR Close #53993

pkozlowski-opensource committed:
refactor(core): extract more query logic from instructions (#54017)

This is a refactor commit that moves more query construction / refresh
logic from the body of instructions into dedicated functions. This is
in preparation for the signal-based query instructions.

PR Close #54017

pkozlowski-opensource committed:
refactor(core): introduce instructions for view queries as signals (#54017)

This commit adds new instructions to support view queries as signals.

PR Close #54017

josephperrott committed:
refactor: migrate bazel, benchpress, elements and misc to prettier formatting (#53995)

Migrate formatting to prettier for bazel, benchpress, elements and misc from clang-format

PR Close #53995

JeanMeche committed:
refactor(devtools): Add hydration informations (#53910)

This commit adds hydration informations to the devtools.
* List of hydrated/hydrated components
* Shows hydration overlays
* Shows hydration errors for NG0500, 501 & 502

PR Close #53910

JeanMeche committed:
refactor(devtools): remove date utility. (#53997)

We can use the `toIsoString()` function instead !

PR Close #53997

JeanMeche committed:
refactor(devtools): migrate to standalone (#53998)

Migrated with the schematics and cleanup by hand.

PR Close #53998

dianjuar committed:
docs: remove ngx-deploy-npm from deployers table (#53999)

PR Close #53999

JeanMeche committed:
docs: update version compatibility table for aio & adev (#54008)

PR Close #54008

d-codez committed:
docs: correct grammar in event handling documentation (#54009)

PR Close #54009

ilirbeqirii committed:
docs: fix typo at component template metadata property (#54015)

Rename the component metadata property, templateUrl to template in the reactive forms tutorial step

PR Close #54015

DivyaSudagoni committed:
docs: move import statement of ngmodule from standalone component to module code #53952 (#54016)

PR Close #54016

pkozlowski-opensource committed:
refactor(core): introduce onDirty notification on QueryList (#54017)

This refactoring expands the QueryList such that we can add onDirty
callback to be invoked when a given query gets marked as dirty during
view insertion / removal. This mechanism is needed for signal-based
queries.

PR Close #54017

devversion committed:
refactor(compiler-cli): additional diagnostics for signal-based queries (#54019)

This commit introduces three additional diagnostics for queries:

- If a query (either using decorator or signal-based) is declared on a
  static class member, a diagnostic is raised.
- If a signal-based query is mixed with a query decorator, a diagnostic
  is raised. Similar to signal inputs.
- If a singal-based query is also declared in the directive/component
  class decorator metadata, a diagnostic is raised.

PR Close #54019

devversion committed:
refactor(compiler): support JIT for signal-based queries (#54019)

Similar to signal-based inputs, we support signal-based queries in JIT
by expecting a decorator to be added. This is a consequence of the
design, given that JIT requires query declaration information before
the class is initialized- but ironically there is no way to collect this
information without instantiating the class.

A JIT transform in the Angular CLI will automatically generate these
decorators for testing.

PR Close #54019

atscott committed:
fix(router): revert commit that replaced `last` helper with native `Array.at(-1)` (#54021)

While `Array.at` is technically supported in all browsers we officially
support, the change was needlessly breaking without any real benefit.

PR Close #54021

atscott committed:
refactor(core): race rAF and setTimeout in zoneless scheduler (#54023)

Update zoneless scheduler to run change detection after the first of
either `requestAnimationFrame` or `setTimeout` callbacks execute.

PR Close #54023

dylhunn committed:
refactor(compiler): ICUs should roll up to the root i18n block for application (#54026)

Previously, if an ICU was inside a nested i18n root, it would use the nested root to calculate whether it should be applied. Now, we use the root i18n block.

PR Close #54026

yjaaidi committed:
test(core): fix test with false-negative risk (#54025)

change assertion to handle the case where the function doesn't throw

PR Close #54025

ChellappanRajan committed:
refactor(forms): Specify `Provider` type for `DEFAULT_VALUE_ACCESSOR` (#54029)

Change from `any` to `Provider` type for enhanced type safety and clarity

PR Close #54029

devversion committed:
fix(compiler-cli): support jumping to definitions of signal-based inputs (#54053)

This fixes the definitions for signal-based inputs in the language
service and type checking symbol builder.

Signal inputs emit a slightly different output. The output works well
for comppletion and was designed to affect language service minimally.
Turns out there is a small adjustment needed for the definition symbols.

PR Close #54053

devversion committed:
refactor(compiler): extract call and construct signatures of interfaces (#54053)

This adds initial support for extracting and rendering call and construct
signatures of classes, like within the new `InputFunction` for signal
inputs.

For now, signatures are a rare occasion and represented as class member
entries. In the future we might consider exposing this via its own entry
type, and field on the class/interface entry.

PR Close #54053

vladboisa committed:
docs: fix & update MDN Shadow API link examples (#54040)

PR Close #54040

maxalmonte14 committed:
docs: fix typo in Custom directives section (#54044)

There was a grammatically incorrect sentence in the What is Angular page, Custom directives section. This change makes it more correct.
PR Close #54044

devversion committed:
build: ensure zone.js typing test is hermetic (#54048)

Currently the ZoneJS typing tests executes outside of Bazel, as a legacy
artifact of the monorepo merging (as it seems - not ideal at all).

Looks like this test relies on its own node modules, that were NOT
locked using a yarn lock file. This commit adds one, and specifically
locks it to a `@types/node` version that does not include the most
recent patch release (which seemingly introduced a breaking change)
that causes issues with TypeScript's lib checking.

Whenever we perform lock file maintenance in the future, we have the
following options:

- Consider disabling lib checking via `skipLibCheck` for this test. This
  may be acceptable.
- Continue locking the node version,
- Waiting for chokidar to comply with the new signature
- Waiting for the breaking change to be rolled back.

Culprit change:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/68300

PR Close #54048

devversion committed:
refactor(core): avoid clang-format obfuscating `InputFunction.required` (#54053)

clang-format seems to have problems with the call signature for
`input.required`. This commit works around the formatting issues that
obfuscate the signature. Users will actually see similar output when
they are looking for the `input` function definition of `@angular/core`.

PR Close #54053

devversion committed:
feat(docs-infra): show overloads of methods in API overview (#54053)

This enables us to show overloads of methods in the API overview. This
is useful for e.g. showing the various signatures of the signal input
function, or for signal-based queris.

There seems to be some issues with the length of the `InputFunction`
overloads. There is some line wrapping that doesn't make it _super_
readable but this is an unrelated problem to this change, but rather
a question of UI / API representation in the angular.io site.

PR Close #54053

devversion committed:
ci: adding devversion to reviewers for compiler/core (#54053)

I've been working on framework parts and compiler since pre-Ivy, and
helped with Ivy, runtime and compiler. Adding myself as a reviewer to
ease future work and to help with review load.

PR Close #54053

dylhunn committed:
refactor(compiler): Support externally provided defer deps fns (#54043)

In #53591, Andrew added local compliation support for defer blocks. However, this requires the ability to emit pre-generated static defer deps functions. We now also support that feature in Template Pipeline.

PR Close #54043

pkozlowski-opensource committed:
docs: release notes for the v17.1.1 release (#54056)

PR Close #54056

alxhub committed:
release: cut the v17.2.0-next.0 release

dylhunn committed:
feat(compiler): Add a TSConfig option `useTemplatePipeline` (#54057)

The Template Pipeline is a brand new backend for the Angular compiler, replacing `TemplateDefinitionBuilder`. It generates the Ivy instructions corresponding to an input template (or host binding). The Template Pipeline has an all-new design based on an intermediate representation compiled over many phases, which will allow us to experiment with compiler changes more easily in the future.

With this commit, the template pipeline can now be enabled in any project via the `useTemplatePipeline` TSConfig option. However, it is still disabled by default.

PR Close #54057

dylhunn committed:
refactor(compiler): Fix defer deps fn duplicate names in Template Pipeline (#54060)

Previously, defer deps fns names were only prefixed with the component name, meaning that distinct deps fns in the same component would produce a name collision. Now, we take into account the entire template function name when naming inner deps fns.

PR Close #54060

crisbeto committed:
refactor(compiler): distinguish two-way bindings in the AST (#54065)

During the template parsing stage two-way bindings are split up into a property and event binding. All the downstream code treats these binding the same as their one-way equivalents. For some future work we'll have to distinguish between the two so these changes update the `BoundElementProperty.type` and `ParsedEvent.type` to include a `TwoWay` type. All existing call-sites have been updated to treat `TwoWay` the same as `Property`/`Regular`, but more specialized logic will be added in the future.

PR Close #54065

crisbeto committed:
fix(core): add toString implementation to signals (#54002)

Since signals are function, currently stringifying them reveals the implementation of the function. This can lead to confusion since it contains internal implementation details. These changes add static `toString` function to address the issue.

**Note:** it's tempting to have `toString` output the actual value of the signal, but that would encourage users not to call the function which will be problematic in the long run. That's why these changes are using a static string instead.

PR Close #54002

JeanMeche committed:
docs: remove mention of providedIn module. This feature has been deprecated. (#54072)

PR Close #54072

bencodezen committed:
docs: add documentation and script for local docs dev environment (#54038)

PR Close #54038

angular-robot committed:
docs: update Angular CLI help [main] (#54075)

Updated Angular CLI help contents.

PR Close #54075

bencodezen committed:
ci: add ben to reviewers for adev files (#54078)

For ADEV changes going forward, Ben will be helping the team to review all PRs associated with this project.

PR Close #54078

crisbeto committed:
refactor(core): signals toString improvements (#54079)

Follow-up to #54002 that:
* Remove the `toString` implementation from the `primitives`.
* Guards the `toString` with `ngDevMode` and prints out the value.

PR Close #54079

thePunderWoman committed:
ci: update adev approvers list (#54081)

This temporarily updates the adev approvers list before we decide how we will handle adev content reviews going forward.

PR Close #54081

jessicajaniuk reviewed (approved):
LGTM!

jessicajaniuk commented:
@nikvarma can you fetch and rebase this PR to main? There's a failing CI test, and it is fixed in latest main. A rebase should fix it.

cexbrayat committed:
fix(core): change defer block fixture default behavior to playthrough (#54088)

This is a followup to #53956

The default behavior needs to be changed in `TestBedCompiler` as well to have an effect.

PR Close #54088

crisbeto committed:
fix(migrations): error in standalone migration when non-array value is used as declarations in TestBed (#54122)

Adds some logic to skip over `TestBed.configureTestingModule` calls where the `declarations` aren't initialized to an array. We can't migrate these cases, because test migrations don't have access to the Angular compiler. Previously the migration would throw a runtime error.

PR Close #54122

amilamen committed:
docs: remove link (#54090)

PR Close #54090

fhoekstra committed:
docs(router): fix variable reference in text (#54099)
PR Close #54099

santoshyadavdev committed:
docs: add link for control flow migration (#53375)

PR Close #53375

mlz11 committed:
docs: fix path to housing location css file in lesson 03 (#54106)

PR Close #54106

strauss02 committed:
docs: update TypeScript version support details for Angular v15 and v16 (#54131)

Clarify the range of TypeScript versions supported in Angular v15 and v16 upgrade guides. The previous documentation's phrasing led to potential misunderstandings, as it suggested broader compatibility than actually supported. This change prevents errors by explicitly stating the maximum TypeScript version for each Angular version, ensuring developers have clear guidance during upgrades.

PR Close #54131

swami-sanapathi committed:
docs: update input/output guide links (#54135)

PR Close #54135

JesseZomer committed:
docs: fix shareReplay refcount default (#54145)

according to the official rxjs docs the refcount default is false. See https://rxjs.dev/api/operators/shareReplay

Co-authored-by: Andrew Scott <atscott01@gmail.com>

PR Close #54145

MarkTechson committed:
docs: Update component-overview.md (#52282)

Refined the text a bit
PR Close #52282

dylhunn committed:
fix(compiler): Fix the template pipeline option (#54148)

The `useTemplatePipeline` option was not properly threaded through the component handler.

PR Close #54148

josephperrott committed:
refactor: migrate common to prettier formatting (#54150)

Migrate formatting to prettier for common from clang-format

PR Close #54150

josephperrott committed:
refactor: migrate docs, examples, private, service worker and upgrade to prettier formatting (#54163)

Migrate formatting to prettier for docs, examples, private, service worker and upgrade from clang-format

PR Close #54163
